fix(release): macOS .app sign order fails v9.8.0 DMG build

## Summary

v9.8.0 リリースワークフロー (`Build DMG installer (macOS)`) の `Sign app bundle` ステップが codesign エラーで exit 1 となり、DMG ビルドが失敗した。`upload-release` と `publish-npm` は `build-dmg` に依存しているため連鎖失敗し、ドラフトリリースが publish されず npm publish も行われなかった。

## 再現

1. `chore(release): v9.8.0` を main にマージ
2. release.yml の `Build DMG installer (macOS)` ジョブが起動
3. `Sign app bundle` ステップで以下のエラー:

    ```text
    dist/GWT.app/Contents/MacOS/gwt: replacing existing signature
    dist/GWT.app/Contents/MacOS/gwt: code object is not signed at all
    In subcomponent: /Users/runner/work/gwt/gwt/dist/GWT.app/Contents/MacOS/gwtd
    ##[error]Process completed with exit code 1.
    ```

Actions run: <https://github.com/akiojin/gwt/actions/runs/24813816226/job/72623938601>

## 原因

`.github/workflows/release.yml` の `Sign app bundle` ステップが CFBundleExecutable (`gwt`) を helper (`gwtd`) より先に `codesign --sign` している。`GWT.app/Contents/MacOS/` 配下で bundle の primary executable を先に署名すると、codesign はバンドル内の他 subcomponent (`gwtd`) が既に署名済みであることを要求し、未署名のため `code object is not signed at all` で失敗する。

## 修正方針

helper バイナリ → CFBundleExecutable → バンドル、の順で `codesign --sign` を呼ぶ。つまり `gwtd` を `gwt` より先に署名するように順序を入れ替える。

- `.github/workflows/release.yml` の `Sign app bundle` ステップで `gwtd` / `gwt` のコマンド順を swap
- `tasks/lessons.md` に codesign 順序の再発防止メモを追加

## 受け入れ条件

- [ ] `Sign app bundle` ステップが exit 0 で通る
- [ ] `Build DMG installer (macOS)` が artifact を upload する
- [ ] `upload-release` が全 artifact (Linux/Windows/macOS portable/MSI/DMG) を draft release に追加する
- [ ] draft release が `--draft` 解除され publish される
- [ ] `publish-npm` が v9.8.0 を npm に publish する

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -348,9 +348,9 @@ jobs:
         run: |
           set -euo pipefail
           codesign --force --options runtime --timestamp --sign "$GWT_CODESIGN_IDENTITY" \
-            dist/GWT.app/Contents/MacOS/gwt
-          codesign --force --options runtime --timestamp --sign "$GWT_CODESIGN_IDENTITY" \
             dist/GWT.app/Contents/MacOS/gwtd
+          codesign --force --options runtime --timestamp --sign "$GWT_CODESIGN_IDENTITY" \
+            dist/GWT.app/Contents/MacOS/gwt
           codesign --force --options runtime --timestamp --sign "$GWT_CODESIGN_IDENTITY" \
             dist/GWT.app
           codesign --verify --deep --strict --verbose=4 dist/GWT.app

--- a/tasks/lessons.md
+++ b/tasks/lessons.md
@@ -1,5 +1,29 @@
 # Lessons Learned
 
+## 2026-04-23 — release: macOS `.app` 内の CFBundleExecutable を他バイナリより先に codesign しない
+
+### 事象
+
+`release.yml` の `Build DMG installer (macOS)` で `Sign app bundle` ステップが
+`dist/GWT.app/Contents/MacOS/gwt: code object is not signed at all`
+`In subcomponent: .../MacOS/gwtd` と出して exit 1。結果として v9.8.0 の DMG ビルドが
+失敗し、`upload-release` / `publish-npm` まで連鎖失敗してドラフトリリースが公開されなかった。
+
+### 原因
+
+`GWT.app/Contents/MacOS/` に CFBundleExecutable (`gwt`) と helper (`gwtd`) が
+並んで置かれているとき、CFBundleExecutable を先に `codesign --sign` すると
+codesign はバンドルの他の subcomponent (この場合 `gwtd`) が既に署名済みで
+あることを要求し、未署名だと "code object is not signed at all" で失敗する。
+
+### 再発防止策
+
+1. `.app` バンドル内を個別に署名する場合、helper バイナリ → CFBundleExecutable → バンドル
+   の順で `codesign --sign` を呼ぶ。順序を逆にしない。
+2. もしくは `codesign --force --deep --sign` を `.app` 1 回だけ呼ぶ。
+3. 新しい helper バイナリを `.app/Contents/MacOS/` に追加するときは、`release.yml` の
+   署名順を必ず見直す。
+
 ## 2026-04-22 — verify: Windows GUI smoke は `browser URL` 出力だけで成功扱いしない
 
 ### 事象

